src/Symfony/Component/HttpClient/Internal/ClientState.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Internal;
+
+/**
+ * Internal representation of the client state.
+ *
+ * @author Alexander M. Turek <me@derrabus.de>
+ *
+ * @internal
+ */
+class ClientState
```

Comment:
final?

Reply:
Since there are two classes extending this one, setting it final would be a bad idea.

Reply:
abstract? or inline per case and avoid a common type.

Reply:
this is internal

Reply:
@ro0NL This is the common part of the state that is required by the shared traits. I could have made that class abstract, but then I would need to create an empty `MockClientState` that would just extend the abstract class.